### DISCUSSION TITLE
Invalid email address (but it is valid!)

### DISCUSSION THREAD

**Discussion by stefanti** [PHPMailer/PHPMailer#2928] [General]
I have found problems when trying to send emails to non "conventional" addresses, this happens for example if I send messages to subscribe or unsubscribe from Aruba mailing addresses.
Example: These addresses are not accepted by PHPMailer but are correct addresses:
verylongmailinglistname-unsubscribe-john=gmail.com@exampledomain.com
or
verylongmailinglistname-subscribe-john=gmail.com@exampledomain.com
or
working.group.7.8.july-subscribe-john=gmail.com@exampledomain.com

These very long or unusual email addresses generate sending errors, if they are shorter the email is sent correctly

**Comment by Synchro** [comment]
"non-conventional" is still simply invalid if it does not conform to RFCs. If you have non-conforming addresses that work, it's your lucky day, but anything using them will be unreliable, so it's worth avoiding. [RFC5321 says that the local part can be a maximum of 64 bytes long](https://www.rfc-editor.org/rfc/rfc5321#section-4.5.3.1.1), so if you're exceeding that, you're simply asking for trouble, in accordance with [Postel's law](https://devopedia.org/postel-s-law). Mind you, it has not escaped my notice that GitHub sends messages that break the local part length limit.

At what point are these actually failing? These are all standard VERP addresses, nothing unusual. All of these examples pass PHPMailer's validator (which is really just a wrapper for PHP's built-in `filter_var`). For example, this returns `true`:

    var_dump(PHPMailer::validateAddress('verylongmailinglistname-unsubscribe-john=gmail.com@exampledomain.com'));

We need more info to go on to diagnose this.
